packages/share_plus_linux/test/share_plus_linux_test.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:share_plus_linux/share_plus_linux.dart';
+import 'package:url_launcher_platform_interface/url_launcher_platform_interface.dart';
+import 'package:url_launcher_platform_interface/link.dart';
+
+void main() {
+  test('url encoding is correct', () {
+    final mock = MockUrlLauncherPlatform();
+    UrlLauncherPlatform.instance = mock;
+
+    ShareLinux().share('foo&bar', subject: 'bar&foo');
+
+    expect(mock.url, 'mailto:?subject=bar%26foo&body=foo%26bar');
+  });
+
+  test('throws when url_launcher can\'t launch uri', () async {
+    final mock = MockUrlLauncherPlatform();
+    mock.canLaunchMockValue = false;
+    UrlLauncherPlatform.instance = mock;
+
+    expect(() async => await ShareLinux().share('foo bar'), throwsException);
+  });
+}
+
+class MockUrlLauncherPlatform extends UrlLauncherPlatform {
```

Comment:
thanks for using fakes instead of mockito!

Reply:
Yeah, mockito is really not that nice to use with nullsafety :(